src/Accounts/Accounts/Rest/InvokeAzRestCommand.cs

```diff
@@ -0,0 +1,169 @@
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Common.Authentication;
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using Microsoft.Azure.Commands.Common.Authentication.Models;
+using Microsoft.Azure.Commands.Profile.Models;
+using Microsoft.Azure.Commands.Profile.Properties;
+using Microsoft.Azure.Commands.ResourceManager.Common;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Management.Automation;
+using System.Net.Http;
+using System.Reflection;
+using System.Threading;
+using Microsoft.Azure.Internal.Common;
+using Microsoft.Rest.Azure;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+
+namespace Microsoft.Azure.Commands.Profile.Rest
+{
+    [Cmdlet("Invoke", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "Rest", DefaultParameterSetName = ByPath, SupportsShouldProcess = true), OutputType(typeof(PSHttpResponse))]
+    public class InvokeAzRestCommand : AzureRMCmdlet
+    {
+        #region Parameter Set
+
+        public const string ByPath = "ByPath";
+        public const string ByParameters = "ByParameters";
+
+        #endregion
+
+        #region Parameter
+
+        [Parameter(ParameterSetName = ByParameters, Mandatory = false, HelpMessage = "Target Subscription Id")]
+        [ValidateNotNullOrEmpty]
+        public string SubscriptionId { get; set; }
+
+        [Parameter(ParameterSetName = ByParameters, Mandatory = false, HelpMessage = "Target Resource Group Name")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(ParameterSetName = ByParameters, Mandatory = false, HelpMessage = "Target Resource Provider Name")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceProviderName { get; set; }
+
+        [Parameter(ParameterSetName = ByParameters, Mandatory = false, HelpMessage = "List of Target Resource Type")]
+        [ValidateNotNullOrEmpty]
+        public string[] ResourceType { get; set; }
+
+        [Parameter(ParameterSetName = ByParameters, Mandatory = false, HelpMessage = "list of Target Resource Name")]
+        [ValidateNotNullOrEmpty]
+        public string[] Name { get; set; }
+
+        [Parameter(ParameterSetName = ByPath, Mandatory = true, HelpMessage = "Target Path")]
+        [ValidateNotNullOrEmpty]
+        public string Path { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "Api Version")]
```

Comment:
Is it mandatory? Can we allow user append API on query path?

Reply:
I prefer leave this as an independent parameter